expr/func_derivative.go

```diff
@@ -0,0 +1,63 @@
+package expr
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/grafana/metrictank/api/models"
+	schema "gopkg.in/raintank/schema.v1"
+)
+
+type FuncDerivative struct {
+	in GraphiteFunc
+}
+
+func NewDerivative() GraphiteFunc {
+	return &FuncDerivative{}
+}
+
+func (s *FuncDerivative) Signature() ([]Arg, []Arg) {
+	return []Arg{
+		ArgSeriesList{val: &s.in}}, []Arg{ArgSeriesList{}}
+}
+
+func (s *FuncDerivative) Context(context Context) Context {
+	return context
```

Comment:
FWIW, i had originally envisioned the context mechanism to allow things like getting an earlier point by adjusting the `from` query time, so that we can start off our first point with a valid one instead of NaN. but that infrastructure isn't quite place yet, and graphite doesn't try to be this fancy anyway, so we can leave this as is. and maybe someday revisit.